java/core/src/java/org/apache/orc/OrcFile.java

```diff
@@ -231,6 +232,33 @@ public static Reader createReader(Path path,
     void preFooterWrite(WriterContext context) throws IOException;
   }
 
+  public static enum BloomFilterVersion {
+    // Include both the BLOOM_FILTER and BLOOM_FILTER_UTF8 streams for string
+    // and decimal columns.
+    ORIGINAL("original"),
+    // Only include the BLOOM_FILTER_UTF8 for string and decimal columns.
+    // See ORC-101
+    UTF8("utf8");
+
+    private final String id;
+    private BloomFilterVersion(String id) {
+      this.id = id;
+    }
+
+    public String toString() {
+      return id;
+    }
+
+    public static BloomFilterVersion fromString(String s) {
+      for (BloomFilterVersion version: values()) {
+        if (version.id.equals(s)) {
+          return version;
+        }
+      }
+      throw new IllegalArgumentException("Unknown BloomFilterVersion " + s);
```

Comment:
I don't see how. To get the value, the code will always do this conversion after getting the string from OrcConf, so the error will be caught coming out.